overlord/configstate/configcore/experimental_test.go

```diff
@@ -96,3 +100,39 @@ func (s *experimentalSuite) TestFilesystemOnlyApplyValidationFails(c *C) {
 	tmpDir := c.MkDir()
 	c.Assert(configcore.FilesystemOnlyApply(classicDev, tmpDir, conf), ErrorMatches, `experimental.refresh-app-awareness can only be set to 'true' or 'false'`)
 }
+
+func (s *experimentalSuite) TestCleanExperimentalFlags(c *C) {
+	o := overlord.Mock()
+	s.state = o.State()
+	hookMgr, err := hookstate.Manager(s.state, o.TaskRunner())
+	c.Assert(err, IsNil)
+
+	s.state.Lock()
+	t := config.NewTransaction(s.state)
+	// Mock an old feature that got out of experimental but still has the config set
+	c.Assert(t.Set("core", "experimental.old-stale-flag", true), IsNil)
+	// Example of an existing experimental flag
+	c.Assert(t.Set("core", "experimental.parallel-instances", true), IsNil)
```

Comment:
This test could maybe override the acceptable experimental flags, so that when parallel-instances becomes non-experimental one day, this test doesn't start failing. Not that big of a deal though.

Reply:
Good point, Thanks! updated